### PR TITLE
Aggregate kicking

### PR DESCRIPTION
**Pull request by tanho63**

### CODE REVIEW

**Comment by tanho63** [comment]
Thought about rewriting stuff to data.table...but got lazy and decided that adding .data$ to everything required less brainpower for right now

File: R/aggregate_kicking_stats.R [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      "team" = "posteam",
+      "player_name" = "kicker_player_name",
+      "player_id" = "kicker_player_id",
+      "kick_distance",
+      "field_goal_attempt",
+      "extra_point_attempt",
+      "field_goal_result",
+      "extra_point_result"
+    ) %>%
+    dplyr::group_by(.data$season,.data$week,.data$season_type,.data$team,.data$player_name,.data$player_id) %>%
```
Comment (mrcaseb) [inline R/aggregate_kicking_stats.R:35]: I wouldn’t group by name and ID here as the names sometimes change during the season or from one season to the next. Instead group by ID only and compute the name with `custom_mode()`

File: R/aggregate_kicking_stats.R [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      "desc",
+      "player_name" = "kicker_player_name",
+      "player_id" = "kicker_player_id",
+      "kick_distance",
+      "field_goal_attempt",
+      "extra_point_attempt",
+      "field_goal_result",
+      "extra_point_result"
+    ) %>%
+    dplyr::group_by(.data$season,.data$week,.data$season_type,.data$team,.data$player_name,.data$player_id) %>%
```
Comment (mrcaseb) [inline R/aggregate_kicking_stats.R:91]: Same like line 35

File: R/aggregate_kicking_stats.R [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      gwfg_blocked = sum(.data$field_goal_result == "blocked", na.rm = TRUE),
+    ) %>%
+    dplyr::ungroup()
+
+  full_kicks <- base_kicks %>%
+    dplyr::left_join(
+      game_winners,
+      by = c("season", "week", "season_type", "team", "player_name", "player_id")) %>%
+    dplyr::mutate(
+      dplyr::across(dplyr::starts_with("gwfg"),tidyr::replace_na,0)
```
Comment (mrcaseb) [inline R/aggregate_kicking_stats.R:106]: We have avoided using dplyr >= 1.0 functions to be compatible with older versions. I think I would stick to that for one more season so we can’t use stuff like `across` or `relocate`

**Review by mrcaseb — COMMENTED** [review]
In general I tend to add a `weekly` argument like in the player stats function. 
If `weekly = FALSE` we summarize across the complete pbp df. Otherwise we do it by season and week. Any thoughts @guga31bb?

**Comment by guga31bb** [comment]
> In general I tend to add a `weekly` argument like in the player stats function.
> If `weekly = FALSE` we summarize across the complete pbp df. Otherwise we do it by season and week. Any thoughts @guga31bb?

Yep agreed

**Comment by tanho63** [comment]
I'm not exactly sure of the value of having the three options (week, season, entire range) and was thinking we should gradually move to making summarizing beyond week up to the user. This is a bit more consistent with what's stored in our data repos, and subsequently with nflreadr etc

You're right re: dplyr 1.0.0, probably won't have time to really clean this up til this weekend or so and I might find the energy to rewrite with datatable by then,,,,

**Comment by atungate** [comment]
@tanho63 et al., thanks for all the work you do to make this package so great!

So, I needed to compute kicking stats and created a function very similar to this PR. After seeing this PR, I updated my function / added most of your columns and matched your column names. Would y'all prefer I submit an issue or PR with the code for you to review? I am sure you are busy working on higher priority things right now but I think this would be great to incorporate.

**Comment by tanho63** [comment]
@atungate Hiya - glad to hear from you! You're more than welcome to fork my PR branch and/or make your own PR that incorporates this one. 

We implemented this in a github action/automation level as `nflreadr::load_player_stats(stat_type = "kicking")` - if you can aim for replicating that dataframe (or improve on) that would be a good place to check against.

**Comment by atungate** [comment]
Ah, very cool!! Ok, sounds good. I can submit tonight. However, quick question, is it needed anymore if y'all have implemented it via GitHub actions? Or is this PR still open to expose a function that works with the current season PBP data? I only ask to make sure I do not waste your time submitting a PR for a solved issue :)

**Comment by tanho63** [comment]
Yeah, we do want to eventually include the parsing of pbp to kicking as a package function (rather than only as a GH script) but the intermediate step sufficed and we haven’t gone back to fixing

**Comment by tanho63** [comment]
Superseded by #381

**Pull request closed without merging**